protocols/cosi/packets.go

```diff
@@ -1,14 +1,104 @@
-package protocol
+package cosi
 
 import (
+	"errors"
+
+	"github.com/dedis/cothority/network"
 	"github.com/dedis/cothority/sda"
 	"github.com/dedis/crypto/abstract"
 )
 
-// This files defines the structure we use for registering to the channels by
-// SDA.
+func init() {
+	sda.RegisterProtocolIO(Name, func() sda.ProtocolIO {
+		return new(ProtocolIO)
+	})
+}
+
+const (
+	// AnnouncementPhase is the ID of the Announcement message
+	AnnouncementPhase uint32 = 1
+	// CommitmentPhase  is the ID of the Commitment message
+	CommitmentPhase = 2
+	// ChallengePhase is the ID of the Challenge message
+	ChallengePhase = 3
+	// ResponsePhase is the ID of the Response message
+	ResponsePhase = 4
+)
+
+// ProtocolPacketID is the network.PacketTypeID of the CoSi ProtocolPacket
+var ProtocolPacketID = network.RegisterPacketType(ProtocolPacket{})
+
+// ProtocolPacket is the main message for the CoSi protocol which includes
+// every information that the CoSi protocol might need.
+type ProtocolPacket struct {
+	Phase uint32
+
+	Info *sda.OverlayMessage
```

Comment:
Mhhh, I'd like the name to represent that it is only a packet, a name, a set of information received, and not a structure which can *do* things such as `Overlay`, so `OverlayMessage` maybe ?